lib/perl/Genome/InstrumentData/Solexa.pm

```diff
@@ -745,14 +745,15 @@ sub _convert_trimmer_to_sx_commands {
                 return;
             }
         } elsif ( $trimmer_params =~ /--\S+?[= ]\S+/ ) {
-            while($trimmer_params =~ /--(\S+)?[= ](\S+)/g) {
-                my ($key, $value) = ($1, $2);
-                #for compatibility with pre-SX far processing profiles
-                if($trimmer_name eq 'far') {
-                    next if $key eq 'format';
-                }
-
-                push @params, $key => $value;
+            my @tokens = split(/\s\-\-/, $trimmer_params);
+            $tokens[0] =~ s/^\-\-//;
+            for my $token ( @tokens ) {
+                my ($key, $value) = split(/\s/, $token, 2);
```

Comment:
You might be able to use [Text::ParseWords::shellwords][1] or similar instead of `split`.  For that matter, you might be able to use [Getopt::Long][2] to parse the whole thing.  If you want to use the OO interface you can use `shellwords` first; that's all `GetOptionsFromString` does.

[1]: http://perldoc.perl.org/Text/ParseWords.html
[2]: http://perldoc.perl.org/Getopt/Long.html#Object-oriented-interface